### PR TITLE
Added support for wire transfers

### PR DESCRIPTION
**Pull request by ADRIANDLT**
Added support to be able to transfer money from one user account to another account from a different user (Wire Transfers)

### CODE REVIEW

**Comment by RestartDK** [comment]
Please resolve conflicts before merge @ADRIANDLT

File: backend/iebank_api/routes.py [hunk 1]
```diff
@@ -445,10 +452,43 @@ def transfer_money():
...
+    from_account = db.session.get(Account, from_account_id)
+    to_account = db.session.query(Account).filter_by(account_number=to_account_number).first()
+
+    if not from_account or not to_account:
+        return jsonify({"error": "Invalid account details"}), 400
 
     try:
+        # We´re reusing the BankTransfer class for wire transfers
+        # In the future if a wire transfer requires a routing number or bank id 
+            # then we will need to create a new class named BankWireTransfer
```
Comment (RestartDK) [inline backend/iebank_api/routes.py:491]: Please remove this comment

File: frontend/src/api.js [hunk 2]
```diff
@@ -58,8 +58,7 @@ export const authService = {
 };
 
 export const accountService = {
-  async getAccounts() {
-    // alert("getAccounts");
+  async getUserAccounts() {
```
Comment (RestartDK) [inline frontend/src/api.js:61]: rename this back to `getAccounts()`

File: frontend/src/components/AppAccounts.vue [hunk 3]
```diff
@@ -268,9 +290,9 @@ export default {
     handleTransferComplete() {
       this.shouldRefreshAccounts = true;
     },
-    async getAccounts() {
+    async fetchAccounts() {
       try {
-        const response = await accountService.getAccounts();
+        const response = await accountService.getUserAccounts();
```
Comment (RestartDK) [inline frontend/src/components/AppAccounts.vue:295]: Rename to `getAccounts()`

File: frontend/src/components/Transfer.vue [hunk 4]
```diff
@@ -105,7 +105,7 @@ export default {
     async fetchAccounts() {
       try {
         //alert("fetching accounts");
-        const response = await accountService.getAccounts();
+        const response = await accountService.getUserAccounts();
```
Comment (RestartDK) [inline frontend/src/components/Transfer.vue:108]: Rename this to `getAccounts`

File: frontend/src/components/WireTransfer.vue [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    },
+    methods: {
+      async fetchAccounts() {
+        try {
+          //alert("fetching accounts");
+          const response = await accountService.getUserAccounts();
+          console.log(response);
+          this.accounts = response.accounts;
+          //alert(respond.accounts[0].account_number);
+          //alert(this.accounts[0].account_number);
```
Comment (RestartDK) [inline frontend/src/components/WireTransfer.vue:107]: Remove the comments

```suggestion
          const response = await accountService.getUserAccounts();
          console.log(response);
          this.accounts = response.accounts;
```

**Review by RestartDK — CHANGES_REQUESTED** [review]
Ok so I put down what you have to change in the code. But some other stuff

Can you instead of manually entering the account code show all the possible accounts you could send it to and who owns it? Because right now you have the `/accounts/all` route that is redundant in your backend.

lmk @ADRIANDLT

**Comment by ADRIANDLT** [comment]
> Ok so I put down what you have to change in the code. But some other stuff
> 
> Can you instead of manually entering the account code show all the possible accounts you could send it to and who owns it? Because right now you have the `/accounts/all` route that is redundant in your backend.
> 
> lmk @ADRIANDLT

Usually in a real bank app/website when you are doing bank wire-transfers you manually type the account number/IBAN so you have the user who is receiving their money give their account number to the sender. The only case where you have a dropdown menu with all the accounts that you can send money to, is when you are transferring money between user account´s/your own accounts.

**Review by RestartDK — APPROVED** [review]
Looks good now

**Pull request merged**